Tribler/community/popular/community.py

```diff
@@ -0,0 +1,501 @@
+        """
+        if peer and peer not in self.get_peers():
+            self.logger.error(ERROR_UNKNOWN_PEER)
+            return
+
+        packet = self.create_message_packet(MSG_CHANNEL_HEALTH_RESPONSE, payload)
+        self.broadcast_message(packet, peer=peer)
+
+    def send_torrent_info_request(self, infohash, peer):
+        """
+        Method to request information about a torrent with given infohash to a peer.
+        """
+        if peer not in self.get_peers():
+            self.logger.error(ERROR_UNKNOWN_PEER)
+            return
+
+        info_request = TorrentInfoRequestPayload(infohash)
+        packet = self.create_message_packet(MSG_TORRENT_INFO_REQUEST, info_request)
+        self.broadcast_message(packet, peer=peer)
+
+    def send_torrent_info_response(self, infohash, peer):
+        """
+        Method to send information about a torrent with given infohash to the requesting peer.
+        """
+        if peer not in self.get_peers():
+            self.logger.error(ERROR_UNKNOWN_PEER)
+            return
+
+        db_torrent = self.content_repository.get_torrent(infohash)
+        info_response = TorrentInfoResponsePayload(infohash, db_torrent['name'], db_torrent['length'],
+                                                   db_torrent['creation_date'], db_torrent['num_files'],
+                                                   db_torrent['comment'])
+        packet = self.create_message_packet(MSG_TORRENT_INFO_RESPONSE, info_response)
+        self.broadcast_message(packet, peer=peer)
+
+    def send_content_info_request(self, content_type, request_list, limit=25, peer=None):
+        cache = self.request_cache.add(SearchRequest(self.request_cache, content_type, request_list))
+        self.logger.info("Sending search request query:%s, identifier:%s", request_list, cache.number)
+
+        content_request = ContentInfoRequest(cache.number, content_type, request_list, limit)
+        packet = self.create_message_packet(MSG_CONTENT_INFO_REQUEST, content_request)
+
+        if peer:
+            self.broadcast_message(packet, peer=peer)
+        else:
+            for connected_peer in self.get_peers():
+                self.broadcast_message(packet, peer=connected_peer)
+
+    def send_content_info_response(self, peer, identifier, content_type, response_list):
+        num_results = len(response_list)
+        current_index = 0
+        page_num = 1
+        while current_index < num_results:
+            serialized_results, current_index, page_size = self.pack_sized(response_list, MAX_PACKET_PAYLOAD_SIZE,
+                                                                           start_index=current_index)
+            if not serialized_results:
+                self.logger.info("Item too big probably to fit into package. Skipping it")
+                current_index += 1
+            else:
+                pagination = Pagination(page_num, page_size, num_results, more=current_index == num_results)
```

Comment:
This code is related to my comment about sending a single UDP message for each torrent/channel vs packing multiple payloads together in a single UDP message.